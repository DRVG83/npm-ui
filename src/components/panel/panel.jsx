import React from 'react';

import './panel.scss';

export class Panel extends React.Component {
  render() {
    return (
      <div 
        className={`${this.props.className} panel-body ${this.props.fullHeight ? 'panel--fullHeight' : 'panel--growingHeight'}`}
      >
        {this.props.children}
      </div>
    );
  }
}

export class ScrollingContent extends React.Component {
  render() {
    return (
      <div className={`${this.props.className} panel--scrollingContent`}>
        {this.props.children}
      </div>
    );
  }
}