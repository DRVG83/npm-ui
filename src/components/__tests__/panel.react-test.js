import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { Panel, ScrollingContent } from "../panel/panel";

test("Panel should render", () => {
  const component = renderer.create(
    <Panel className="some-class-name" fullHeight={true}>
      Content
    </Panel>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Panel should contain the custom className", () => {
  const customClassName = "some-class-name";
  const wrapper = shallow(<Panel className={customClassName}>Content</Panel>);
  expect(wrapper.find("div").hasClass(customClassName)).toBe(true);
});

test("Panel should render in full height", () => {
  const wrapper = shallow(<Panel fullHeight={true}>Content</Panel>);
  expect(wrapper.find("div").hasClass("panel--fullHeight")).toBe(true);
});

test("Panel should render in growing height", () => {
  const wrapper = shallow(<Panel fullHeight={false}>Content</Panel>);
  expect(wrapper.find("div").hasClass("panel--growingHeight")).toBe(true);
});

test("ScrollingContent should render", () => {
  const component = renderer.create(
    <ScrollingContent className="some-class-name">Content</ScrollingContent>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("ScrollingContent should contain the custom className", () => {
  const customClassName = "some-class-name";
  const wrapper = shallow(
    <ScrollingContent className={customClassName}>Content</ScrollingContent>
  );
  expect(wrapper.find("div").hasClass(customClassName)).toBe(true);
});
