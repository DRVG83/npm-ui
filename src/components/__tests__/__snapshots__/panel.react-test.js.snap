// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render 1`] = `
<div
  className="some-class-name panel-body panel--fullHeight"
>
  Content
</div>
`;
